hcid-js/README.md

```diff
@@ -0,0 +1,44 @@
+# HCID-JS
+
+Generate Holochain IDs in javascript. A thin wrapper around rust compiled to WASM
+
+## Installation
+
+This package is distributed via npm and can be installed using
+`npm install @holochain/hcid`
+
+## Usage
+
+This module exports a class called `Encoding` which can be used to construct an encoding for the different types of Holochain identifiers. Each of these identifiers are given a three character prefix:
+- AgentID (from signing key) : 'hcs'
+- ...
+
+Depending on if you are using the module in node.js or the browser the calling syntax is slightly different. This is because in the browser the WASM must be asynchronously compiled and instantiated to prevent blocking the main thread. As a result all of the constructor returns a promises in the browser but not in node.
+
+```javascript
+
+const publicKey = [...] // UInt8Array of bytes of public key
+
+const enc = new Encoding('hcs') // node.js
+const enc = await new Encoding('hcs') // browser
```

Comment:
these need `0`s ... `new Encoding('hcs0')` we've got a version bit in there we can update when we get a 5 bit reed-solomon encoder in rust.